android/engine/src/main/java/org/smartregister/fhircore/engine/util/ReflectionUtil.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2021-2023 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.engine.util
+
+import kotlin.reflect.KClass
+import kotlin.reflect.full.callSuspend
+import kotlin.reflect.full.declaredMemberFunctions
+import kotlin.reflect.full.declaredMemberProperties
+import kotlin.reflect.jvm.isAccessible
+
+fun <T : Any> KClass<T>.callFunction(method: String, vararg args: Any) =
+  this.declaredMemberFunctions
+    .firstOrNull { it.name == method }
+    ?.apply { isAccessible = true }
+    ?.call(this, args)
+
+suspend fun <T : Any> KClass<T>.callSuspendFunctionOnField(
```

Comment:
@maimoonak Is there any alternative to using reflection? How would using reflection handle API changes on the relevant classes?

Reply:
Yeah. I am creating the issues and PRs on SDK as well. This is supposed to be changed whenever SDK has functionality available. There was no way to implement this without heavily coping functions and extension from SDK. This involved multiple classes and methods which would have become difficult to manage overtime.